index.js

```diff
@@ -797,3 +790,16 @@ function preappend (blocks) {
     this.encryption.encrypt(offset + i, blocks[i], fork)
   }
 }
+
+function ensureEncryption (core, opts) {
+  if (opts.encryptionKey) {
+    // Only override the block encryption if its either not already set or if
+    // the caller provided a different key.
+    if (
+      !core.encryption ||
+      !b4a.equals(core.encryption.key, opts.encryptionKey)
+    ) {
+      core.encryption = new BlockEncryption(opts.encryptionKey, core.key)
+    }
+  }
```

Comment:
since this is a function now we can simplify

if (!opts.encryptionKey) return
if (core.encryption) return
if (b4a.equals(...)) return